src/view/position.js

```diff
@@ -176,6 +176,25 @@ export default class Position {
 	}
 
 	/**
+	 * Returns a {@link module:engine/view/node~Node} which is a common ancestor for both positions.
+	 *
+	 * @param {module:engine/view/position~Position} position
+	 * @returns {module:engine/view/node~Node|null}
+	 */
+	getCommonAncestor( position ) {
+		const ancestorsA = this.getAncestors();
```

Comment:
The second issue – you didn't pass `options.parentFirst` too so how can you iterate from `i==0` (leafs)? What if one ancestors array is longer than the other?